tool/tctl/common/admin_action_test.go

```diff
@@ -180,6 +180,160 @@ func (s *AdminActionTestSuite) TestRole() {
 	})
 }
 
+func (s *AdminActionTestSuite) TestClusterAuthPreference() {
+	t := s.T()
+	ctx := context.Background()
+
+	originalAuthPref, err := s.authServer.GetAuthPreference(ctx)
+	require.NoError(t, err)
+
+	createAuthPref := func() error {
+		// To maintain the current auth preference for each test case, get
+		// the current auth preference from config and change it to dynamic.
+		authPref, err := s.authServer.GetAuthPreference(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		authPref.SetOrigin(types.OriginDynamic)
+		return s.authServer.SetAuthPreference(ctx, authPref)
+	}
+
+	getAuthPref := func() (types.Resource, error) {
+		return s.authServer.GetAuthPreference(ctx)
+	}
+
+	resetAuthPref := func() error {
+		return s.authServer.SetAuthPreference(ctx, originalAuthPref)
+	}
+
+	t.Run("ResourceCommands", func(t *testing.T) {
+		s.testResourceCommand(t, ctx, resourceCommandTestCase{
+			resource:       originalAuthPref,
+			resourceCreate: createAuthPref,
+			resourceDelete: resetAuthPref,
```

Comment:
Maybe more appropriately named `resourceCleanup`? Consider revising the naming and/or adding commentary explanations to this `resourceCommandTestCase` object.